modules/gumgumBidAdapter.js

```diff
@@ -0,0 +1,171 @@
+    pu: utils.getTopWindowUrl(),
+    ce: utils.cookiesAreEnabled(),
+    dpr: topWindow.devicePixelRatio || 1
+  }
+  return browserParams
+}
+
+function getWrapperCode(wrapper, data) {
+  return wrapper.replace('AD_JSON', window.btoa(JSON.stringify(data)))
+}
+
+// TODO: use getConfig()
+function _getDigiTrustQueryParams() {
+  function getDigiTrustId () {
+    var digiTrustUser = (window.DigiTrust && window.DigiTrust.getUser) ? window.DigiTrust.getUser(DT_CREDENTIALS) : {};
+    return (digiTrustUser && digiTrustUser.success && digiTrustUser.identity) || '';
+  };
+
+  let digiTrustId = getDigiTrustId();
+  // Verify there is an ID and this user has not opted out
+  if (!digiTrustId || (digiTrustId.privacy && digiTrustId.privacy.optout)) {
+    return {};
+  }
+  return {
+    'dt': digiTrustId.id
+  };
+}
+
+/**
+ * Determines whether or not the given bid request is valid.
+ *
+ * @param {BidRequest} bid The bid params to validate.
+ * @return boolean True if this is a valid bid, and false otherwise.
+ */
+function isBidRequestValid (bid) {
+  const {
+    params,
+    adUnitCode
+  } = bid;
+
+  switch (true) {
+    case !!(params.inScreen): break;
+    case !!(params.inSlot): break;
+    default:
+      utils.logWarn(`[GumGum] No product selected for the placement ${adUnitCode}, please check your implementation.`);
+      return false;
+  }
+  return true;
+}
+
+/**
+ * Make a server request from the list of BidRequests.
+ *
+ * @param {validBidRequests[]} - an array of bids
+ * @return ServerRequest Info describing the request to the server.
+ */
+function buildRequests (validBidRequests) {
+  const bids = [];
+  utils._each(validBidRequests, bidRequest => {
+    const timeout = config.getConfig('bidderTimeout');
+    const {
+      bidId,
+      params = {},
+      transactionId
+    } = bidRequest;
+    const data = {}
+
+    if (params.inScreen) {
+      data.t = params.inScreen;
+      data.pi = 2;
+    }
+    if (params.inSlot) {
+      data.si = parseInt(params.inSlot, 10);
+      data.pi = 3;
+    }
+
+    bids.push({
+      id: bidId,
+      tmax: timeout,
+      tId: transactionId,
+      pi: data.pi,
+      sizes: bidRequest.sizes,
+      url: BID_ENDPOINT,
+      method: 'GET',
+      data: Object.assign(data, _getBrowserParams(), _getDigiTrustQueryParams())
+    })
+  });
+  return bids;
+}
+
+/**
+ * Unpack the response from the server into a list of bids.
+ *
+ * @param {*} serverResponse A successful response from the server.
+ * @return {Bid[]} An array of bids which were nested inside the server.
+ */
+function interpretResponse (serverResponse, bidRequest) {
+  const bidResponses = []
+  const serverResponseBody = serverResponse.body
+  const {
+    ad: {
+      price: cpm,
+      id: creativeId,
+      markup
+    },
+    cw: wrapper
+  } = serverResponseBody
+  let isTestUnit = (bidRequest.data && bidRequest.data.pi === 3 && bidRequest.data.si === 9)
+  let [width, height] = bidRequest.sizes[0]
```

Comment:
ad unit sizes may be defined as a double array (`[[300, 250]]`) or single array (`[300, 250]`). This will break if the size is defined as a single array, but `utils.parseSizesInput` can help account for either case. Something like `utils.parseSizesInput(bidRequest.sizes)[0].split('x')` if you only want the first size in the list